### PR TITLE
patch additional data

### LINKED ISSUE #221: Implement methods needed for patchAdditionalDataEndpointV2 in Server
Implement following methods in order to introduce PATCH operation on partitioning's additional data.

```scala
case class AdditionalDataDTO(
  data: Map[String, Option[AdditionalDataItemDTO]],
)

object AdditionalDataDTO {
  implicit val encodeAdditionalDataDTO: Encoder[AdditionalDataDTO] = deriveEncoder[AdditionalDataDTO]
  implicit val decodeAdditionalDataDTO: Decoder[AdditionalDataDTO] = deriveDecoder[AdditionalDataDTO]
}

case class AdditionalDataPatchDTO(
  data: Map[String, AdditionalDataItemDTO]
)

object AdditionalDataPatchDTO {
  implicit val encoderAdditionalDataPatchDTO: Encoder[AdditionalDataPatchDTO] = deriveEncoder[AdditionalDataPatchDTO]
  implicit val decoderAdditionalDataPatchDTO: Decoder[AdditionalDataPatchDTO] = deriveDecoder[AdditionalDataPatchDTO]
}

case class AdditionalDataItemDTO(
  value: String,
  author: String
)

object AdditionalDataItemDTO {
  implicit val encoderAdditionalDataItem: Encoder[AdditionalDataItemDTO] = deriveEncoder[AdditionalDataItemDTO]
  implicit val decoderAdditionalDataItem: Decoder[AdditionalDataItemDTO] = deriveDecoder[AdditionalDataItemDTO]
}

// in Endpoints
protected val patchPartitioningAdditionalDataEndpointV2
    : PublicEndpoint[(Long, AdditionalDataPatchDTO), ErrorResponse, SingleSuccessResponse[
      AdditionalDataDTO
    ], Any] = {
    apiV2.patch
      .in(V2Paths.Partitionings / path[Long]("partitioningId") / V2Paths.AdditionalData)
      .in(jsonBody[AdditionalDataPatchDTO])
      .out(statusCode(StatusCode.Ok))
      .out(jsonBody[SingleSuccessResponse[AdditionalDataDTO]])
  }

// in Routes.scala
createServerEndpoint[
        (Long, AdditionalDataPatchDTO),
        ErrorResponse,
        SingleSuccessResponse[AdditionalDataDTO]
      ](
        patchPartitioningAdditionalDataEndpointV2,
        { case (partitioningId: Long, additionalDataPatchDTO: AdditionalDataPatchDTO) =>
          PartitioningController.patchPartitioningAdditionalDataV2(partitioningId, additionalDataPatchDTO)
        }
      )

// in PartitioningControllerImpl
override def patchPartitioningAdditionalDataV2(
    partitioningId: Long,
    additionalData: AdditionalDataPatchDTO
  ): IO[ErrorResponse, SingleSuccessResponse[AdditionalDataDTO]]

// in PartitioningServiceImpl
override def patchAdditionalData(
    partitioningId: Long,
    additionalData: AdditionalDataPatchDTO
  ): IO[ServiceError, AdditionalDataDTO]

// in PartitioningRepositoryImpl
override def patchAdditionalData(
    partitioningId: Long,
    additionalData: AdditionalDataDTO
  ): IO[DatabaseError, AdditionalDataDTO]
```

Also implement SQL function for the PATCH operation. The function should create or update records and return ALL additional data related to given partitioning.

Also implement fa-db class for the SQL code.

Remove obsolete code related to CreateOrUpdateAdditionalData.

### PR DESCRIPTION
**Pull request by salamonpavel**
Introduces PATCH endpoint for additional data
Closes #221

### CODE REVIEW

**Comment by salamonpavel** [comment]
Release notes:
- Introduces PATCH endpoint for additional data

File: server/src/main/scala/za/co/absa/atum/server/api/database/runs/functions/CreateOrUpdateAdditionalData.scala [hunk 1]
```diff
@@ -17,31 +17,42 @@
...
-    extends DoobieSingleResultFunctionWithStatus[AdditionalDataSubmitDTO, Unit, Task](values =>
+    extends DoobieMultipleResultFunctionWithAggStatus[CreateOrUpdateAdditionalDataArgs, Option[
+      AdditionalDataItemFromDB
+    ], Task](args =>
       Seq(
-        fr"${PartitioningForDB.fromSeqPartitionDTO(values.partitioning).asJson}",
-        fr"${values.additionalData.map { case (k, v) => (k, v.orNull) }}",
-        fr"${values.author}"
+        fr"${args.partitioningId}",
+        fr"${args.additionalData.data}",
```
Comment (lsulak) [inline server/src/main/scala/za/co/absa/atum/server/api/database/runs/functions/CreateOrUpdateAdditionalData.scala:40]: I'll take a deeper and proper look later, but why it's not allowed to have Option there?

Comment (salamonpavel) [reply server/src/main/scala/za/co/absa/atum/server/api/database/runs/functions/CreateOrUpdateAdditionalData.scala:40]: We don't want to allow 'empty' values for PATCH operation. That's been discussed.

File: database/src/main/postgres/runs/V1.5.16__create_or_update_additional_data.sql [hunk 2]
```diff
@@ -14,80 +14,80 @@
...
+    IN i_by_user TEXT,
+    OUT status INTEGER,
+    OUT status_text TEXT,
+    OUT o_ad_name TEXT,
+    OUT o_ad_value TEXT,
+    OUT o_ad_author TEXT
+) RETURNS SETOF record AS
 $$
--------------------------------------------------------------------------------
+    -------------------------------------------------------------------------------
```
[the remark this answers is not in the document]
Comment (salamonpavel) [reply database/src/main/postgres/runs/V1.5.16__create_or_update_additional_data.sql:27]: I do not recall adding this whitespace, probably automatic formatting was applied ... Commit below accepted.

File: server/src/main/scala/za/co/absa/atum/server/api/database/runs/functions/CreateOrUpdateAdditionalData.scala (hunk 1 above)
Comment (lsulak) [reply server/src/main/scala/za/co/absa/atum/server/api/database/runs/functions/CreateOrUpdateAdditionalData.scala:40]: Okay, thanks for the explanation

File: database/src/main/postgres/runs/V1.5.16__create_or_update_additional_data.sql [hunk 3]
```diff
@@ -14,80 +14,81 @@
...
```
Comment (benedeki) [inline database/src/main/postgres/runs/V1.5.16__create_or_update_additional_data.sql:27]:
```suggestion
-------------------------------------------------------------------------------
```

**Review by benedeki — APPROVED** [review]
LGTM

**Review by lsulak — APPROVED** [review]

**Pull request merged**